### DISCUSSION TITLE
qql query sample

### DISCUSSION THREAD

**Discussion by danielvaz21** [finos/TimeBase-CE#288] [General]
Hi [@alex-karpovich](https://github.com/alex-karpovich),
 
I couldn't find qql query to create view in documentation. Can you please help us with the sample qql or sample code where I can find examples for same.

Regards
Daniel Vaz

**Comment by danielvaz21** [comment]
Hi [@alex-karpovich](https://github.com/alex-karpovich),

Can you please help us with the sample qql or sample code where I can find examples for same.

Regards
Daniel Vaz

**Comment by alex-karpovich** [comment]
Greetings, @danielvaz21 

Do you need create `view` via Java API?

**Reply by danielvaz21** [reply]
Greetings [@alex-karpovich](https://github.com/alex-karpovich),

We need to create a view via Python API

**Reply by Romkisel** [reply]
Hi, @danielvaz21, please, try this example of creating a view: https://github.com/epam/TimebaseWS/blob/main-1.1/samples/python/create_view_sample.py

**Reply by danielvaz21** [reply]
Greeting @Romkisel @alex-karpovich 

We have tried doing the same with the request library and are able to create the views, however we are looking for the qql query that we can use to create the view using python tbapi library.

**Comment by ankitsoin** [comment]
Hi @alex-karpovich, also please help us knowing few more queries:

1. If we keep feature LIVE = ON then is there any performance issue on base table as it has to keep view updated all the time.
2. At what frequency does the view gets refreshed to keep in sync with base table.

**Reply by alex-karpovich** [reply]
1. When LIVE = ON, then it will process when live data is coming into source stream, of course it will cost CPU
2. we have pool of workers (default pool size = 4) that processing all views in parallel

**Comment by ankitsoin** [comment]
Hi @alex-karpovich,

I have created one stream and loaded 100 records in it. Then view is created on top of the same stream using simple query "select * from stream_obd" with LIVE = ON. 
Currently our view is displaying all 100 records.

Now I have loaded 100 more records in "stream_obd". When I query my view again **these new 100 records were not visible**.
Our understanding is it should show 200 records from the base stream, but it isn't.

Is this an expected behavior?

_Note:
Currently view is being created using Web admin and stream is created by QQL using python client.
We have tried loading data in both with and without spaces of the stream._

**Reply by alex-karpovich** [reply]
fixed in https://github.com/finos/TimeBase-CE/commit/239f4b3d8cfa594cc3b6fb67e6560f2dd125a50c

**Reply by ankitsoin** [reply]
Ok great. Thanks.
Please let us know when we are going to deploy this change

**Comment by danielvaz21** [comment]
Hi [@alex-karpovich](https://github.com/alex-karpovich),

Can you please help us with how we checked the memory utilization for a Space/View in timebase.

Regards
Daniel Vaz

**Reply by alex-karpovich** [reply]
lets me add metrics into timebase server to be able to monitor system load, like we have for EE version
https://kb.timebase.info/docs/development/sys-metrics
